sdk/go/common/workspace/plugins.go

```diff
@@ -353,14 +357,54 @@ func newGithubSource(url *url.URL, name string, kind PluginKind) (*githubSource,
 }
 
 func (source *githubSource) newHTTPRequest(url, accept string) (*http.Request, error) {
-	req, err := buildHTTPRequest(url, fmt.Sprintf("token %s", source.token))
+	var token string
+	if source.token != "" {
+		token = fmt.Sprintf("token %s", source.token)
+	}
+	req, err := buildHTTPRequest(url, token)
 	if err != nil {
 		return nil, err
 	}
 	req.Header.Set("Accept", accept)
 	return req, nil
 }
 
+func (source *githubSource) retryHTTP(
+	getHTTPResponse func(*http.Request) (io.ReadCloser, int64, error),
+	req *http.Request) (io.ReadCloser, int64, error) {
+
+	delay := 80 * time.Millisecond
+	for attempt := 0; ; attempt++ {
+		resp, length, err := getHTTPResponse(req)
+		if err == nil {
+			return resp, length, nil
+		}
+
+		// Retry 403 rate limits
+		if err, ok := err.(*downloadError); ok && err.code == 403 {
```

Comment:
We should probably use errors.As here so that if in the future, getHTTPResponse wraps the error with more context, this still works.
```suggestion
		// Retry 403 rate limits
		var downErr *downloadError
		if errors.As(err, &downErr) && downErr.code == 403 {
```